.github/workflows/ci.yml

```diff
@@ -0,0 +1,109 @@
+name: CI
+on:
+  pull_request:
+    types: [opened, synchronize, reopened, ready_for_review]
+  push:
+    branches: [master]
+    tags:
+      - '*'
+
+jobs:
+  docs:
+   runs-on: ubuntu-latest
+   if: github.event.pull_request.draft == false
+
+   steps:
+     - name: Checkout
+       uses: actions/checkout@v2
+
+     - name: Validate docs
+       run: npx elm make --docs=/tmp/docs.json
+
+  lint:
+   runs-on: ubuntu-latest
+   if: github.event.pull_request.draft == false
+
+   steps:
+     - name: Checkout
+       uses: actions/checkout@v2
+
+     - name: Setup Node
+       uses: actions/setup-node@v1
+       with:
+        node-version: '14'
+
+     - name: Get yarn cache directory path
+       id: yarn-cache-dir-path
+       run: echo "::set-output name=dir::$(yarn cache dir)"
+
+     - uses: actions/cache@v1
+       id: yarn-cache
+       with:
+        path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+        key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+        restore-keys: |
+          ${{ runner.os }}-yarn-
+
+     - name: Install dependencies
+       run: yarn install
+
+     - name: Run linter
+       run: npx elm-format --validate src/ examples/
+
+     - name: Run analyse
+       run: npx elm-analyse
+
+  example1:
```

Comment:
Shouldn't we run the examples together in the same job? maybe in a more automated way, something like run `npx elm make` for every file in the examples folder 🤔 , otherwise, if we add a new example, we will likely forgot to add it to CI

Reply:
![last try CI logs](https://user-images.githubusercontent.com/1368952/101018001-9db98d00-3549-11eb-9059-fe7c8dcbbc11.png)

Reply:
Fancy stuff 👍